tests/validation/tests/v3_api/test_cluster_templates.py

```diff
@@ -714,6 +716,224 @@ def test_cluster_template_create_with_monitoring():
+    cluster_template = \
+        admin_client.create_cluster_template(name=cluster_template_name,
+                                             description="test-template",
+                                             members=members)
+    clusterTemplateId = cluster_template.id
+    # Create Cluster template revision
+    revision_name = random_test_name("revision")
+    cluster_template_revision = \
+        admin_client.create_cluster_template_revision(
+            name=revision_name,
+            clusterConfig=cluster_config,
+            clusterTemplateId=clusterTemplateId,
+            enabled="true")
+
+    standard_user_client = \
+        get_client_for_token(
+            user_token["standard_user"]["token"])
+
+    # Verify the standard user does not have permission to edit/update the
+    # template as he has only user access
+    members = [{"type": "member", "accessType": "read-only",
+                "userPrincipalId": principalId},
+               {"type": "member", "accessType": "read-only",
+                "groupPrincipalId": "*"}]
+    with pytest.raises(ApiError) as e:
+        standard_user_client.update(cluster_template,
+                                    name="sjtest",
+                                    description="test-template",
+                                    members=members)
+        assert e.value.error.status == 403
+        assert e.value.error.code == "PermissionDenied"
+
+    # Verify the standard user does not have permission to delete the
+    # revision as he has only user/read-only access
+    with pytest.raises(ApiError) as e:
+        standard_user_client.delete(cluster_template_revision)
+        assert e.value.error.status == 403
+        assert e.value.error.code == "PermissionDenied"
+
+    # Verify the standard user does not have permission to delete the
+    # template as he has only user access
+    with pytest.raises(ApiError) as e:
+        standard_user_client.delete(cluster_template)
+        assert e.value.error.status == 403
+        assert e.value.error.code == "PermissionDenied"
+
+
+def validate_cluster_with_template(client, cluster,
+                                   intermediate_state="provisioning",
+                                   check_intermediate_state=True,
+                                   nodes_not_in_active_state=[],
+                                   k8s_version="", userToken=USER_TOKEN):
+    '''
+    In this method, we are checking cluster state, verifying state of workloads
+    in system project. For user workloads, we are just checking the state. We
+    are skipping the kubectl verification for user workloads because of this
+    issue: https://github.com/rancher/rancher/issues/27788
+    Hence this method is introduced locally in test_cluster_templates.py and
+    we are not using validate_cluster method from common.py
+    '''
+
+    # Allow sometime for the "cluster_owner" CRTB to take effect
+    time.sleep(5)
```

Comment:
Isn't there something in `test_rbac` to validate the cluster owner? Instead of a fixed wait.

Reply:
Done